samples/network/httptest/source/http.c

```diff
@@ -0,0 +1,319 @@
+#include <http/https.h>
+#include <ssl/ssl.h>
+#include <net/net.h>
+#include <sysmodule/sysmodule.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#define HTTP_YES		1
+#define HTTP_NO 		0
+#define HTTP_SUCCESS 	1
+#define HTTP_FAILED	 	0
+
+#define HTTP_USER_AGENT "Mozilla/5.0 (PLAYSTATION 3; 1.00)"
+
+typedef struct
+{
+    void* http_pool;
+    void* ssl_pool;
+    httpsData* caList;
+    void* cert_buffer;
+} t_http_pools;
+
+static t_http_pools http_pools;
+u8 cancel = HTTP_NO;
+
+static char getBuffer[64*1024];
+
+
+int http_init(void)
+{
+    int ret;
+	u32 cert_size=0;
+	u8 module_https_loaded=HTTP_NO;
+	u8 module_http_loaded=HTTP_NO;
+	u8 module_net_loaded=HTTP_NO;
+	u8 module_ssl_loaded=HTTP_NO;
+	
+	u8 https_init=HTTP_NO;
+	u8 http_init=HTTP_NO;
+	u8 net_init=HTTP_NO;
+	u8 ssl_init=HTTP_NO;
+
+	//init
+	ret = sysModuleLoad(SYSMODULE_NET);
+	if (ret < 0) {
+		printf("Error : sysModuleLoad(SYSMODULE_NET) HTTP_FAILED (%x)", ret);
+		ret=HTTP_FAILED;
+		goto end;
+	} else module_net_loaded=HTTP_YES;
+
+	ret = netInitialize();
+	if (ret < 0) {
+		printf("Error : netInitialize HTTP_FAILED (%x)", ret);
+		ret=HTTP_FAILED;
+		goto end;
+	} else net_init=HTTP_YES;
+
+	ret = sysModuleLoad(SYSMODULE_HTTP);
+	if (ret < 0) {
+		printf("Error : sysModuleLoad(SYSMODULE_HTTP) HTTP_FAILED (%x)", ret);
+		ret=HTTP_FAILED;
+		goto end;
+	} else module_http_loaded=HTTP_YES;
+
+	http_pools.http_pool = malloc(0x10000);
+	if (http_pools.http_pool == NULL) {
+		printf("Error : out of memory (http_pool)");
+		ret=HTTP_FAILED;
+		goto end;
+	}
+
+	ret = httpInit(http_pools.http_pool, 0x10000);
```

Comment:
So libnet manages memory for a user (128KB) but for these other libs you have to have your own memory pool for initialization? Interesting, it'd be nice if there was more consistency between the libs, but at the same time I could understand wanting to define your own memory size for the http pool

Reply:
well, don't forget we are calling the original PS3 libraries, so the rationale here was defined by SCE. I don't know the details so I can't really answer. The original SDK brings the SCE documentation, and I guess some reasons could be found there.